### PR TITLE
Kc/kedro integration advanced

### PR DESCRIPTION
**Pull request by jakubczakon**
* started working on advanced use cases

### CODE REVIEW

**Comment by jakubczakon** [comment]
@kamil-kaczmarek seems like the failing ones are about some random connection lost error and not Kedro pipelines at all.

**Review by kamil-kaczmarek — COMMENTED** [review]
For the most part the code is created by kerdo -> is that right?

There a re also test directories - was that intentional? Do you plan to run these tests, or are they created by kedro automatically?

**Comment by jakubczakon** [comment]
@kamil-kaczmarek yes, the code is mostly Kedro code, the only files where Neptune changes live are:
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog_neptune.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/neptune.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/parameters.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/local/credentials_neptune.yml
*  integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/data/01_raw/iris.csv
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/data/01_raw/iris_v2.csv

I don't plan to run any tests -> they were just included with Kedro.
Actually I'll try and download/run this on clean env without the tests folders just to see if Kedro is fine with it.

**Comment by jakubczakon** [comment]
@kamil-kaczmarek now I added all the files from those kedro projects and adjusted the test run_example.sh to make it work. 
Take a look and tell me what you think.

**Comment by kamil-kaczmarek** [comment]
ok, @jakubczakon let me have a look tomorrow morning

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#    if_exists: 'replace'
+#    # if_exists: 'fail'
+#    # if_exists: 'append'
+#
+# The Data Catalog supports being able to reference the same file using two different DataSet implementations
+# (transcoding), templating and a way to reuse arguments that are frequently repeated. See more here:
+# https://kedro.readthedocs.io/en/stable/05_data/01_data_catalog.html
+#
+# This is a data set used by the "Hello World" example pipeline provided with the project
+# template. Please feel free to remove it once you remove the example pipeline.
```
Comment (kamil-kaczmarek) [inline integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog.yml:43]: I would drop all unnecessary things from this file. It's a bit misleading, as I do not know what is relevant to the example and what is not.

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog_neptune.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#   filepath: data/06_models/clf_model.pkl
+#
+# If you want to log existing Kedro Dataset to Neptune add @neptune to the DataSet name
+#
+# example_iris_data@neptune:
+#   type: kedro_neptune.NeptuneFileDataSet
+#   filepath: data/01_raw/iris.csv
+#
+# You can use kedro_neptune.NeptuneFileDataSet in any catalog including conf/base/catalog.yml
+#
```
Comment (kamil-kaczmarek) [inline integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog_neptune.yml:14]: Seems like the entire config is commented out. I wonder if we actually need it?

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    predictions = {}
+    for name, model in zip(['rf', 'mlp'], [rf_model, mlp_model]):
+        y_pred = model.predict_proba(test_x).tolist()
+        predictions[name] = y_pred
+
+    return predictions
+
+
+def evaluate_models(predictions: dict, test_y: pd.DataFrame,
+                    neptune_run: neptune.run.Handler):
```
Comment (kamil-kaczmarek) [inline integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py:90]: Not sure, but it probably should be 
`neptune_run: neptune.run.Run` -> it's neptune Run object that you use for logging.

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        plot_roc_curve(test_y.idxmax(axis=1), y_pred, ax=ax, title=f'ROC curve {name}')
+        neptune_run['nodes/evaluate_models/plots/plot_roc_curve'].log(fig)
+
+        fig, ax = plt.subplots()
+        plot_precision_recall_curve(test_y.idxmax(axis=1), y_pred, ax=ax, title=f'PR curve {name}')
+        neptune_run['nodes/evaluate_models/plots/plot_precision_recall_curve'].log(fig)
+
+
+def ensemble_models(predictions: dict, test_y: pd.DataFrame,
+                    neptune_run: neptune.run.Handler) -> np.ndarray:
```
Comment (kamil-kaczmarek) [inline integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py:110]: Not sure, but it probably should be
`neptune_run: neptune.run.Run` -> it's neptune Run object that you use for logging.

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example code for the nodes in the example pipeline. This code is meant
+just for illustrating basic Kedro features.
+
+Delete this when you start working on your own Kedro project.
+"""
```
Comment (kamil-kaczmarek) [inline integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py:33]: maybe dropping this

**Review by kamil-kaczmarek — APPROVED** [review]
left few comment, nothing is blocking

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py (hunk 4 above)
Comment (jakubczakon) [reply integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py:110]: it is actualy a handler afaik,

File: integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py (hunk 5 above)
Comment (jakubczakon) [reply integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py:33]: mhm we could -> but as the example starts with the kedro starter that has it and builds on top of it, I think I'd leave as is.

**Pull request merged**